### PR TITLE
Salto 2143: add documentation on timeout request retries

### PR DESCRIPTION
**Pull request by tamtamirr**

### CODE REVIEW

**Comment by tamtamirr** [comment]
Followup to the previous PR: https://github.com/salto-io/salto/pull/3024

**Comment by ori-moisis** [comment]
will we have a log when this timeout happens?
can we track it to see if this actually works?

File: packages/salesforce-adapter/config_doc.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ salesforce {
       maxAttempts = 5
       retryDelay = 5000
       retryStrategy = "NetworkError"
+      timeout = 1000 * 60 * 15
```
Comment (ori-moisis) [inline packages/salesforce-adapter/config_doc.md:24]: pretty sure we do no support math operations in config
did you test this?

File: packages/salesforce-adapter/config_doc.md [hunk 2]
```diff
@@ -208,11 +209,12 @@ For more details see the DeployOptions section in the [salesforce documentation
...
-|----------------|------------------------|------------
-| maxAttempts    | `5`                    | The number of attempts to make for each request
-| retryDelay     | `5000` (5 seconds)     | The time (milliseconds) to wait between attempts
-| retryStrategy  | `NetworkError`         | In which cases to retry. Supported choices: `NetworkError` (retry on network errors), `HttpError` (retry on HTTP 5xx errors), or `HTTPOrNetworkError` (both)
+| Name          | Default when undefined        | Description
+|---------------|-------------------------------|------------
+| maxAttempts   | `5`                           | The number of attempts to make for each request
+| retryDelay    | `5000` (5 seconds)            | The time (milliseconds) to wait between attempts
+| retryStrategy | `NetworkError`                | In which cases to retry. Supported choices: `NetworkError` (retry on network errors), `HttpError` (retry on HTTP 5xx errors), or `HTTPOrNetworkError` (both)
+| timeout       | `1000 * 60 * 15` (15 minutes) | The timeout (milliseconds) on each request retry
```
Comment (ori-moisis) [inline packages/salesforce-adapter/config_doc.md:217]: same comment as above

File: packages/salesforce-adapter/config_doc.md (hunk 1 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/config_doc.md:24]: Yes , tested this and we don't forgot to update the example

File: packages/salesforce-adapter/config_doc.md (hunk 2 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/config_doc.md]: v

**Comment by tamtamirr** [comment]
@ori-moisis 
* We do have log
* I've tested that locally by setting timeout of 1 second, and it did throw timeout.

**Review by ori-moisis — APPROVED** [review]

**Pull request merged**